src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+                exceptionHandler.handleDukeExceptions(e);
+            } catch (Exception e) {
+                exceptionHandler.handleOtherExceptions(e);
+            }
+        }
+    }
+
+    /**
+     * Reads the text entered by the user.
+     *
+     * @param in The Scanner to read user input.
+     * @return line Full line entered by the user.
+     */
+    private static String getUserInput(Scanner in) {
+        String line = in.nextLine();
+        return line;
+    }
+
+    /**
+     * Executes the command as specified by the user input.
+     *
+     * @param userInput Raw input from user.
+     */
+    private static void executeCommand(String userInput) throws DukeException, IOException {
+        final String[] commandTypeAndParams = parser.splitCommandAndArgs(userInput);
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+
+        switch (commandType) {
+        case COMMAND_ADD_TODO:
+            executeAddTodo(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_ADD_DEADLINE:
+            executeAddDeadline(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_ADD_EVENT:
+            executeAddEvent(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_LIST:
+            executeList();
+            break;
+        case COMMAND_MARK_DONE:
+            executeMarkDone(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_DELETE:
+            executeDelete(commandArgs);
+            storage.saveDataToFile(tasks);
+            break;
+        case COMMAND_EXIT:
+            executeExit();
+            break;
+        default:
+            throw new DukeException();
+        }
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Executes exiting program.
+     */
+    private static void executeExit() throws IOException {
+        ui.printGoodBye();
+        System.exit(0);
+    }
+
+    /**
+     * Executes marking task as done.
+     *
+     * @param commandArgs Processed input arguments from user.
+     */
+    private static void executeMarkDone(String commandArgs) {
+        int taskIndex = Integer.parseInt(commandArgs.strip()) - 1;
+        Task currentTask = tasks.getTaskByIndex(taskIndex);
+        currentTask.markAsDone();
+        ui.printConfirmMarkDone(currentTask);
+    }
+
+    private static void executeDelete(String commandArgs) {
+        int taskIndex = Integer.parseInt(commandArgs.strip()) - 1;
+        ui.printConfirmDelete(tasks.getTaskByIndex(taskIndex), tasks.getNumberOfTasks());
+        tasks.removeTaskByIndex(taskIndex);
+    }
+
+    /**
+     * Executes listing tasks.
+     */
+    private static void executeList() {
+        System.out.println(HORIZONTAL_LINE);
+        if (tasks.getNumberOfTasks() > 0) {
```

Comment:
You can make the [happy path](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) more prominent here. Eg. check if number of tasks <= 0, if yes, print the MESSAGE_NO_TASK then `return`. Then the rest of your code to handle the subsequent logic doesn't need to be nested inside `if else` blocks. Similar thing can be done for `executeAddEvent` and some other similar functions here